src/App.tsx

```diff
@@ -47,13 +49,21 @@ export const App = () => {
       window.addEventListener('scroll', refreshJwtToken);
       window.addEventListener('onfocus', refreshJwtToken);
       window.addEventListener('keydown', refreshJwtToken);
+      window.addEventListener('hashchange', setPdfState);
     }
 
     function removeEventListeners() {
       window.removeEventListener('mousemove', refreshJwtToken);
       window.removeEventListener('scroll', refreshJwtToken);
       window.removeEventListener('onfocus', refreshJwtToken);
       window.removeEventListener('keydown', refreshJwtToken);
+      window.removeEventListener('hashchange', setPdfState);
+    }
+
+    function setPdfState() {
+      if (shouldGeneratePdf()) {
+        dispatch(PdfActions.pdfStateChanged());
+      }
     }
```

Comment:
That is one possible way to go about it. Downsides are that in automatic-mode, it will need to fetch the pdf format, which is never used for regular users filling out the form. It is also needed to signal whether preview should be active or not (otherwise you would need to refresh the page after setting the parameter). Preview mode will refetch the pdfformat (in auto mode) on each save because it is essentially dynamics computed on the backend, and it could change based on form data.